icu4c/source/common/uloc.cpp

```diff
@@ -1787,9 +1787,16 @@ uloc_getParent(const char*    localeID,
         i=0;
     }
 
-    if(i>0 && parent != localeID) {
-        uprv_memcpy(parent, localeID, uprv_min(i, parentCapacity));
+    if (i > 0) {
+        if (uprv_strnicmp(localeID, "und", 3) == 0 && (localeID[3] == '\0' || localeID[3] == '_')) {
```

Comment:
Should we check for `localeID[3] == '@'` as well, or are extensions stripped by this point?

Reply:
`uloc_getParent()` is a rather crude function, it will just return the empty string unless it finds an underscore so there is no need to check for any other separator.

But I see now that the condition can be simplified, which might also make it more clear. Done.